### PR TITLE
Feature/module 3 components

### PR DESCRIPTION
**Pull request by UltramarineDev**

### CODE REVIEW

File: netflix-roulette/src/components/App/error-boundary/ErrorBoundaryContainer.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    super(props);
+    this.state = { hasError: false };
+  }
+  
+  static getDerivedStateFromError(error) {
+    return { hasError: true };
+  }
+  
+  componentDidCatch(error, errorInfo) {
+    logError(error, errorInfo);
```
Comment (UltramarineDev) [inline netflix-roulette/src/components/App/error-boundary/ErrorBoundaryContainer.js:19]: need to discuss the correct usage.

File: netflix-roulette/src/components/features/home/constants.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  HORROR: 'HORROR',
+  CRIME: 'CRIME',
+};
+
+export const FILTER_TAGS = {
+  DOCUMENTARY: ['Documentary', 'Adventure'],
+  COMEDY: ['Comedy', 'Family', 'Fantasy'],
+  HORROR: ['Horror', 'Action'],
+  CRIME: ['Crime', 'Action'],
+};
```
Comment (UltramarineDev) [inline netflix-roulette/src/components/features/home/constants.js:14]: Added extra tags for displaying test data.

File: netflix-roulette/src/components/App/App.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React, { Suspense } from 'react';
+
+import ErrorBoundary from './error-boundary';
+import Layout from '../layout';
+
+const Home = React.lazy(() => import('../features/home'));
+const HomeWithLayout = Layout(Home);
+
+//TODO: Implement app default font family, font size
```
Comment (UltramarineDev) [inline netflix-roulette/src/components/App/App.js:9]: I will enhance styling during module 4, 5.

File: netflix-roulette/src/components/App/App.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const App = () => (
+  <>
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/App/App.js]: -

File: netflix-roulette/src/components/features/home/Home.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  moviesCount,
+  tabs,
+  activeTab,
+  onTabChange,
+  sortOptions,
+  onSortOptionChange,
+  sortOptionId,
+  }) => (
+  <>
+   <div className={styles.searchSectionWrapper}>
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/home/Home.js]: Header

File: netflix-roulette/src/components/features/home/Home.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+       <div className={styles.header}>
+         <NetflixRoulette/>
+         <Button type={BUTTON_TYPES.ADD} iconType={BUTTON_ICONS.ADD} label='ADD MOVIE'/>
+       </div>
+       <div className={styles.searchDescription}>find your movie</div>
+       <Search />
+     </div>
+    </div>
+
+    <div className={styles.movieListSection}>
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/home/Home.js:21]: NavBar

File: netflix-roulette/src/components/features/home/HomeContainer.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/home/HomeContainer.js]: PureComponent

File: netflix-roulette/src/components/features/home/HomeContainer.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+  componentDidMount() {
+    this.loadPage();
+  }
+
+  handleTabChange = (activeTab) => {
+    this.setState(({ movies }) => {
+      let filteredMovies = movies;
+
+      //TODO: Maybe optimize filtering
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/home/HomeContainer.js]: Let's move to utils folder

File: netflix-roulette/src/components/features/home/constants.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+export const TABS = {
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/home/constants.js:1]: lower case

File: netflix-roulette/src/components/shared/button/Button.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { BUTTON_TYPES } from '/src/components/shared/constants';
+import styles from './Button.module.scss';
+
+const ButtonComponent = ({ label, type, icon }) => (
+  <>
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/shared/button/Button.js:10]: const className = type === BUTTON_TYPES.ADD ? addButton : searchButton:


className={styles[className]}

File: netflix-roulette/src/components/shared/button/Button.module.scss [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  margin: 0;
+  cursor: pointer;
+  overflow: visible;
+  text-decoration: none;
+  white-space: nowrap;
+  height: 60px;
+  width: 200px;
+  line-height: 40px;
+  font-size: 32px;
+  font-family: "Inv Maison Neue","Maison Neue","Open Sans",open-sans,sans-serif;
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/shared/button/Button.module.scss:13]: cdn 
google fonts

index.html
<link ..../>

File: netflix-roulette/src/components/shared/button/utils.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { BUTTON_ICONS } from '/src/components/shared/constants';
+
+export const getButtonIcon = (icon) => {
+  switch (icon) {
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/shared/button/utils.js:7]: const btnIcon = {
add:  "plus",
remove: "minus"
}

icon = {btnIcon[type]}

File: netflix-roulette/src/components/shared/dropdown/DropdownContainer.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+  static propTypes = {
+    value: PropTypes.string.isRequired,
+    options: PropTypes.arrayOf(PropTypes.PropTypes.shape({
+      id: PropTypes.string,
+      name: PropTypes.string,
+    })),
+    onChange: PropTypes.func.isRequired,
+  };
+
+  state = {
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/shared/dropdown/DropdownContainer.js:16]: const [showOptions, setShowOptions] = useState(false);

setShowOptions(true);

File: netflix-roulette/src/components/shared/search/Search.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import { BUTTON_TYPES } from '/src/components/shared/constants';
+
+import Button from '../button';
+import SearchField from './search-field';
+import styles from './Search.module.scss';
+
+const SearchComponent = () => (
+  <>
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/shared/search/Search.js]: -

File: netflix-roulette/src/components/shared/tab-bar/tab-bar-item/TabBarItem.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import styles from './TabBarItem.module.scss';
+
+const TabBarItemComponent = ({ label, active, onClick }) => (
+  <>
+    {active ? 
+      <a className={styles.activeTab} onClick={onClick}>{label}</a> :
+      <a className={styles.tab} onClick={onClick}>{label}</a>
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/shared/tab-bar/tab-bar-item/TabBarItem.js]:
```
className={clsx(styles.tab, {
[styles.activeTab]: active,
});

```

**Review by omelenetskiy — APPROVED** [review]

File: netflix-roulette/src/components/App/App.js (hunk 4 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/App/App.js]: done.

File: netflix-roulette/src/components/shared/search/Search.js (hunk 14 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/shared/search/Search.js]: done.

File: netflix-roulette/src/components/features/home/Home.js (hunk 6 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/features/home/Home.js:21]: done.

File: netflix-roulette/src/components/features/home/Home.js (hunk 5 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/features/home/Home.js]: done.

**Pull request merged**